core/src/test/java/org/elasticsearch/search/sort/SortParserTests.java

```diff
@@ -47,7 +47,7 @@ public void testGeoDistanceSortParserManyPointsNoException() throws Exception {
         sortBuilder.endArray();
         sortBuilder.field("order", "desc");
         sortBuilder.field("unit", "km");
-        sortBuilder.field("sort_mode", "max");
+        sortBuilder.field("mode", "max");
```

Comment:
This is an example of how things are broken right now, the parser pfm gets set to STRICT in the test, then the parser is added to the context with `reset()`which changes is back to EMPTY (this is the source of the error here, will be fixed with this PR) and then it is never reset.